core/domain/classifier_services.py

```diff
@@ -442,7 +445,17 @@ def load_examples(self, examples):
             xrange(self._num_docs))
 
     def predict_label_for_doc(self, d):
-        """Returns the predicted label from a doc's prediction report."""
+        """Returns the predicted label from a doc's prediction report.
+
+        Classifiers built with less than _DEFAULT_MIN_DOCS_TO_PREDICT will
```

Comment:
Regarding the two sentences of this paragraph:

For the first part, I'm a bit confused on what the value '20' represents here. You mention that if a label is added with less than 20 docs, it probably won't be useful for predicting. I'm not sure I understand why or how it could potentially cause issues with incorrect predictions. I like how the second part of this paragraph gives reasoning why the labels value is exactly 3, but this variable needs more discussion I think.

For the second part, this sounds more like a doc string for the _DEFAULT_MIN_LABELS_TO_PREDICT variable rather than this function. It's discussing implementation details of the function, which aren't as useful to the client code. Perhaps instead mention in this docstring to refer to _DEFAULT_MIN_DOCS_TO_PREDICT and _DEFAULT_MIN_LABELS_TO_PREDICT to ensure the caller code has setup the classifier with at least that many labels and at least that many docs per label. If they haven't, discuss that the DEFAULT_LABEL will be selected instead to guard against incorrect or inaccurate predictions, etc.